lib/index.js

```diff
@@ -99,366 +96,295 @@ exports.Boom = class extends Error {
 
         return this.prototype.isPrototypeOf(instance);
     }
-};
-
 
-exports.isBoom = function (err, statusCode) {
+    reformat(debug = false) {
 
-    return err instanceof Error && !!err.isBoom && (!statusCode || err.output.statusCode === statusCode);
-};
+        this.output.payload = new internals.PayloadObject(this, this.output.statusCode, debug);
+    }
 
+    #apply(data, statusCode, headers, message) {
 
-exports.boomify = function (err, options) {
+        if (data !== undefined) {
+            this.data = data;
+        }
 
-    Hoek.assert(err instanceof Error, 'Cannot wrap non-Error object');
+        if (statusCode) {
+            const numberCode = parseInt(statusCode, 10);
+            if (isNaN(numberCode) || numberCode < 400) {
+                throw new TypeError(`statusCode must be a number (400+): ${statusCode}`);
+            }
 
-    options = options || {};
+            if (message) {
+                this.message = `${message}: ${this.message}`;
+            }
 
-    if (options.data !== undefined) {
-        err.data = options.data;
+            const payload = new internals.PayloadObject(this, numberCode, false);
+            this.output = new internals.BoomOutput(numberCode, payload, headers);
+        }
     }
 
-    if (options.decorate) {
-        Object.assign(err, options.decorate);
-    }
+    static {
+        Object.defineProperty(this.prototype, 'name', { value: 'Boom', writable: true, configurable: true });
```

Comment:
Would it make sense to make this a little bit more "`extend` friendly" ?

```js
class Boom {
  static {
    // Allow child classes to inherit the same behavior
    Object.defineProperty(this.prototype, 'name', {
      configurable: true,
      get() {
        return this.constructor.name
      },
      // Allow overriding instance name
      set(value) {
        Object.defineProperty(this, 'name', {
          value,
          writable: true,
          enumerable: true,
          configurable: true,
        })
      }
    })
  }
}

class BoomChild extends Boom {}

const boom = new Boom()
console.log(boom.name) // Boom
boom.name = "NewName"
console.log(boom.name) // NewName

const child = new BoomChild()
console.log(child.name) // BoomChild

child.name = "NewName"
console.log(child.name) // NewName
```

Reply:
The property can already be extended through `writable: true`, supporting normal usage.

I don't like the inference from the class name, as you suggest. Extending a `Error` does not do this, so it would be somewhat of a surprise. I would just expect to use the regular pattern:

```js
class SubBoom extends Boom {
    name = 'SubBoom';
}
```

Reply:
I have pushed a commit that tests for the expected "name" behaviour.